datautils.py

```diff
@@ -11,140 +11,90 @@ def set_seed(seed):
+        return testenc
+
+
+def get_ptb(nsamples, seqlen, tokenizer, new=False, eval_mode=False):
+    sep = "\n\n" if not new else " "
+    test_split = "validation" if not new else "test"
+
+    if not eval_mode:
+        traindata = load_dataset("ptb_text_only", "penn_treebank", split="train")
+        trainenc = tokenizer(sep.join(traindata["sentence"]), return_tensors="pt")
+        trainloader = []
+        for _ in range(nsamples):
+            i = random.randint(0, trainenc.input_ids.shape[1] - seqlen - 1)
+            j = i + seqlen
+            inp = trainenc.input_ids[:, i:j]
+            tar = inp.clone()
+            tar[:, :-1] = -100
+            trainloader.append((inp, tar))
+        return trainloader
+    else:
+        testdata = load_dataset("ptb_text_only", "penn_treebank", split=test_split)
+        testenc = tokenizer(sep.join(testdata["sentence"]), return_tensors="pt")
+        return testenc
 
 
-def get_loaders(name, custom_data_path=None, nsamples=128, seed=0, seqlen=2048, model_path=""):
+def get_c4(nsamples, seqlen, tokenizer, new=False, eval_mode=False):
+    if not eval_mode:
+        traindata = load_dataset(
+            "allenai/c4", "allenai--c4", data_files={"train": "en/c4-train.00000-of-01024.json.gz"}, split="train"
+        )
+        trainloader = []
+        for _ in range(nsamples):
+            while True:
+                i = random.randint(0, len(traindata) - 1)
+                trainenc = tokenizer(traindata[i]["text"], return_tensors="pt")
+                if trainenc.input_ids.shape[1] >= seqlen:
+                    break
+            i = random.randint(0, trainenc.input_ids.shape[1] - seqlen - 1)
+            j = i + seqlen
+            inp = trainenc.input_ids[:, i:j]
+            tar = inp.clone()
+            tar[:, :-1] = -100
+            trainloader.append((inp, tar))
+        return trainloader
+    else:
+        valdata = load_dataset("allenai/c4", "allenai--c4",
+                               data_files={"validation": "en/c4-validation.00000-of-00008.json.gz"}, split="validation")
+        random.seed(0)
+        if new:
+            valenc = tokenizer(" ".join(valdata[:1100]["text"]), return_tensors="pt")
+            valenc = valenc.input_ids[:, : (256 * seqlen)]
+        else:
+            valenc = []
+            for _ in range(256):
+                while True:
+                    i = random.randint(0, len(valdata) - 1)
+                    tmp = tokenizer(valdata[i]["text"], return_tensors="pt")
+                    if tmp.input_ids.shape[1] >= seqlen:
+                        break
+                i = random.randint(0, tmp.input_ids.shape[1] - seqlen - 1)
+                j = i + seqlen
+                valenc.append(tmp.input_ids[:, i:j])
+            valenc = torch.hstack(valenc)
+
+        return valenc
+
+
+def get_loaders(name, nsamples=128, seed=0, seqlen=2048, model_path="", eval_mode=False):
```

Comment:
Now this function returns either train or validation dataset instead of both. I would suggest to update docstring.

Reply:
Eval mode is not is very common - maybe is_train would be more intuitive?

Reply:
#1 agreed. will update the docstring soon. 
#2 well, this may be the reason to have more common option as default.